server/aesdsocket.c

```diff
@@ -262,173 +559,154 @@ int main(int argc, char *argv[])
+        thread_data->thread_info.thread_complete_success = false;
+        strcpy(thread_data->thread_info.ip4, ip4);
 
-        do{
-            
-            bytes_rx = recv(new_fd, my_buffer + addibuffer, BUF_SIZE-1, 0); 
-            syslog(LOG_DEBUG, "i have received %ld bytes", bytes_rx);
-            if (bytes_rx < 0)
-            {
-                free(my_buffer);
-                perror("server: recv");
-                syslog(LOG_ERR, "recv: error??");
-                exit(EXIT_FAILURE);
-            }
-            addibuffer += bytes_rx; 
-            new_line = strchr(my_buffer, '\n'); //if NULL, keep getting packets
-            if (new_line != NULL)
-            {
-                syslog(LOG_DEBUG, "YAY, slash n found");
-                isPacketValid = true;
-                addibuffer = new_line - my_buffer + 1; 
-                syslog(LOG_DEBUG, "addibuffer size within slash found %ld", addibuffer);
-            }
+        int rc;
 
-            else
-            {
-                my_buffer = realloc(my_buffer, incrementBy+BUF_SIZE);
-                if(my_buffer == NULL)
-                {
-                    syslog(LOG_ERR, "Failed to realloc.");
-                    close(new_fd);
-                    exit(EXIT_FAILURE);
-                }
-                incrementBy += BUF_SIZE;
-                memset(my_buffer+addibuffer, 0 , incrementBy-addibuffer);
-                //addibuffer = 0;
-            }
-        //syslog(LOG_DEBUG, "I am here once only. %d", isPacketValid);
+        if((rc = pthread_create(&thread_data->thread_info.threadid, NULL, threadfunc, (void*) &(thread_data->thread_info))) != 0)
+        {
+            syslog(LOG_ERR, "Failed to pthread_create(), error was %d", rc);
+            perror("Failed to pthread_create()");
+            free(thread_data);
+            close(new_fd);
+            continue;
+        }
 
-        }while (isPacketValid == false);
+        else 
+        {
+            syslog(LOG_DEBUG, "pthread_created successfully. The threadID is %lu. AFD: %d", thread_data->thread_info.threadid, thread_data->thread_info.afd);
+        }
 
-        //new line character has been found, now i want to send?
-        //append happens here, writing
-        if(isPacketValid == true)
-        {   
-            //my_buffer[addibuffer]= '\0';
-            //ssize_t write_size = new_line - my_buffer + 1;
-            syslog(LOG_DEBUG, "PACKET SUCCESSFULLY VALIDATED");
-            nr = write(recvfile_fd, my_buffer, addibuffer);
-            syslog(LOG_DEBUG, "nr is %ld",nr);
-            if (nr != addibuffer)
-            {
-                /*error*/
-                int err1 = errno;
-                syslog(LOG_ERR, "Failed to write bytes: errno -> %d", err1);
-                syslog(LOG_ERR, "error string is %s", strerror(errno));
-                exit(EXIT_FAILURE);
-            }
 
-            syslog(LOG_DEBUG, "Write completed to recvfile_fd");
+        rc = pthread_mutex_lock(&ll_lock);
```

Comment:
@sota6640 I don't know that the lock is serving any purpose here since you are only accessing the list in a single thread.